main.js

```diff
@@ -0,0 +1,379 @@
+}
+
+var toggleAISwitch = 1; //toggle that switches between player and AI
+var toggleAICount = 0;
+function toggleAI() { //toggles whether AI should be on/off
+    toggleAICount = 1 - toggleAICount;
+    $('.title').text("Choose your color and play with the Gods.");
+    if (toggleAICount === 0 ) {
+        $('.title').text("Player One: Choose Your Diety");
+    }
+    playerModeToggle = 1;
+    $('.togglePlayerNumber').text("THREE PLAYER MODE");
+};
+
+//Function to assign image class to players
+function playerColor() {
+    if (playerSwitch === 2) {
+        if (playerOneColor === null) {
+            playerOneColor = $(this).css("background-image");
+        }
+    } else if (playerSwitch === 1) {
+        if (playerTwoColor === null) {
+            playerTwoColor = $(this).css("background-image");
+        }
+    } else if (playerSwitch === 0 && playerModeToggle === 0) {
+        if (playerThreeColor === null) {
+            playerThreeColor = $(this).css("background-image");
+        }
+    }
+    $(this).addClass("gray");
+    $(this).off("click");
+    $(".toggleAI, .togglePlayerNumber").hide();
+    playerSwitch--;
+    if (playerSwitch === 1 && toggleAICount === 0) {
+        $('.title').text("Player Two: Choose Your Diety");
+    } else if (playerSwitch === 0 && toggleAICount === 0) {
+        $('.title').text("Player Three: Choose Your Diety");
+    }
+    if (toggleAICount === 1 && victoryTrigger === 0) { //if AI mode is on, runs function that allows AI to randomly choose color
+        aiSelectColor();
+    };
+    if (playerOneColor != null && playerTwoColor != null && playerThreeColor != null && toggleAICount === 0 || playerOneColor != null && playerTwoColor != null && toggleAICount === 1 || playerOneColor != null && playerTwoColor != null && playerModeToggle === 1) {
+        $('.choose-color-page').hide();
+        $('.game_area').show();
+        $('.gameTitle').text("Player One's Turn");
+        playerSwitch = 2;
+    }
+}
+
+function aiSelectColor() { // Allows AI to pick random color after player one chooses color
+    var colorArray = [$(".red").css("background-image"), $(".blue").css("background-image"), $(".green").css("background-image"), $(".green").css("background-image")];
+    if (playerOneColor != null) {
+        for (var colorCount = 0; colorCount < colorArray.length; colorCount++) {
+            if (playerOneColor === colorArray[colorCount]) {
+                colorArray.splice(colorCount, 1); //removes player one color from colorArray so AI doesn't pick undefined color
+            }
+        }
+        var randomColorNum = Math.floor((Math.random() * colorArray.length ));
+        playerTwoColor = colorArray[randomColorNum];
+    }
+    toggleAISwitch = 1 - toggleAISwitch;
+};
+
+function addToGridArray() { //function that stores respective player's number into gameArray
+    for (var rowCount = gameArray.length-1; rowCount >= 0; rowCount--) {
+        if (gameArray[rowCount][columnNumber] === null) {
```

Comment:
This doesn't account for the possibility that no moves are available in that column. I see two potential problems with this: with just human players, a player could click a fully occupied column, and still end up using their turn, despite not placing a piece. Also, the AI could select a fully occupied column by mistake, and lose their turn as well. You need to have a condition to check if not move was made (maybe a flag within the function's scope?), and exit the function if there was no possible move. You would also need a way for the AI to make another move, if they chose a fully occupied space.